app/public/src/interface.js

```diff
@@ -0,0 +1,193 @@
+(function(){'use strict';}());
+
+$(document).ready(function(){
+  var game = new Game();
+  var roll = new Roll();
+  disableSpareButton();
+
+  $("[type=button]").click(function(){
+    rollBall(this.name);
+  });
+
+  function rollBall(pinsDown){
+    if(roll.rollComplete === true){
+      roll = new Roll();
+      disableSpareButton();
+    }
+    roll.addRoll(parseInt(pinsDown));
+
+    if((roll.rollComplete === false)&&(roll.roll.length === 1)){
+      if(game.frames.length === 10){
+        roll.rollComplete = true;
+      } else
+
+      if(game.frames.length === 11){
+        roll.rollComplete = true;
+      } else {
+        var num = game.currentRollNumber();
+        var num1 = num+1;
+        $('#edit-frame'+num1+'-1').val(roll.roll[0]);
+        disableButtons(roll.roll[0]);
+        enableSpareButton(roll.roll[0]);
+      }
+    }
+     if (roll.rollComplete){
+      $("[id=var1-9]").prop('disabled',true);
+      game.totalScore = 0;
+      game.addFrame(roll);
+      if((game.frames.length === 10)&&(roll.score === 10)){
```

Comment:
This looks like logic that relates to the game rules and therefore belongs in the model.